### DISCUSSION TITLE
Error while making a build on Android

### DISCUSSION THREAD

**Discussion by UmberFunk** [yasirkula/UnityNativeShare#141] [Q&A] [answered]
Hello,

First, I want to say thanks for providing such a good tool!

I am trying to make a build on Android using Unity 2019.4.34, but get these errors I pasted below.

Any help will be highly appreciated.

Assets\Store Assets\P&C UI Gradient Effect\UI Gradient plugin\Scripts\UIGradient.cs(19,36): warning CS0067: The event 'UIGradient.requireRepaint' is never used

> Task :launcher:preBuild UP-TO-DATE
> Task :unityLibrary:preBuild UP-TO-DATE
> Task :unityLibrary:FirebaseApp.androidlib:preBuild UP-TO-DATE
> Task :unityLibrary:FirebaseApp.androidlib:preDebugBuild UP-TO-DATE
> Task :unityLibrary:preDebugBuild UP-TO-DATE
> Task :launcher:prepareLintJar UP-TO-DATE
> Task :unityLibrary:checkDebugManifest UP-TO-DATE
> Task :unityLibrary:FirebaseApp.androidlib:checkDebugManifest
> Task :unityLibrary:processDebugManifest
> Task :unityLibrary:FirebaseApp.androidlib:processDebugManifest
> Task :unityLibrary:packageDebugRenderscript NO-SOURCE
> Task :unityLibrary:FirebaseApp.androidlib:compileDebugAidl NO-SOURCE
> Task :unityLibrary:generateDebugBuildConfig UP-TO-DATE
> Task :unityLibrary:FirebaseApp.androidlib:packageDebugRenderscript NO-SOURCE
> Task :unityLibrary:compileDebugAidl NO-SOURCE
> Task :unityLibrary:FirebaseApp.androidlib:compileDebugRenderscript NO-SOURCE
> Task :unityLibrary:compileDebugRenderscript NO-SOURCE
> Task :unityLibrary:generateDebugResValues UP-TO-DATE
> Task :unityLibrary:generateDebugResources UP-TO-DATE
> Task :unityLibrary:FirebaseApp.androidlib:generateDebugResValues
> Task :unityLibrary:FirebaseApp.androidlib:generateDebugResources
> Task :unityLibrary:packageDebugResources UP-TO-DATE
> Task :unityLibrary:prepareLintJar UP-TO-DATE
> Task :unityLibrary:generateDebugSources UP-TO-DATE
> Task :unityLibrary:mergeDebugShaders UP-TO-DATE
> Task :unityLibrary:compileDebugShaders UP-TO-DATE
> Task :unityLibrary:generateDebugAssets UP-TO-DATE
> Task :unityLibrary:FirebaseApp.androidlib:packageDebugResources
> Task :unityLibrary:FirebaseApp.androidlib:generateDebugRFile
> Task :unityLibrary:FirebaseApp.androidlib:generateDebugBuildConfig
> Task :unityLibrary:FirebaseApp.androidlib:prepareLintJar UP-TO-DATE
> Task :unityLibrary:FirebaseApp.androidlib:generateDebugSources
> Task :unityLibrary:FirebaseApp.androidlib:javaPreCompileDebug
> Task :unityLibrary:FirebaseApp.androidlib:compileDebugJavaWithJavac
> Task :unityLibrary:packageDebugAssets
> Task :unityLibrary:FirebaseApp.androidlib:bundleLibCompileDebug
> Task :unityLibrary:FirebaseApp.androidlib:mergeDebugShaders
> Task :unityLibrary:FirebaseApp.androidlib:compileDebugShaders
> Task :unityLibrary:FirebaseApp.androidlib:generateDebugAssets
> Task :unityLibrary:FirebaseApp.androidlib:packageDebugAssets
> Task :unityLibrary:FirebaseApp.androidlib:bundleLibRuntimeDebug
> Task :unityLibrary:FirebaseApp.androidlib:processDebugJavaRes NO-SOURCE
> Task :unityLibrary:FirebaseApp.androidlib:bundleLibResDebug
> Task :unityLibrary:FirebaseApp.androidlib:mergeDebugJniLibFolders
> Task :unityLibrary:FirebaseApp.androidlib:transformNativeLibsWithMergeJniLibsForDebug
> Task :unityLibrary:FirebaseApp.androidlib:transformNativeLibsWithStripDebugSymbolForDebug
> Task :unityLibrary:FirebaseApp.androidlib:transformNativeLibsWithIntermediateJniLibsForDebug
> Task :launcher:preDebugBuild
> Task :launcher:compileDebugAidl NO-SOURCE
> Task :launcher:compileDebugRenderscript NO-SOURCE
> Task :launcher:checkDebugManifest UP-TO-DATE
> Task :launcher:generateDebugBuildConfig UP-TO-DATE
> Task :launcher:generateDebugSources UP-TO-DATE
> Task :launcher:mainApkListPersistenceDebug UP-TO-DATE
> Task :launcher:generateDebugResValues UP-TO-DATE
> Task :launcher:generateDebugResources UP-TO-DATE
> Task :unityLibrary:generateDebugRFile
> Task :unityLibrary:javaPreCompileDebug
> Task :unityLibrary:compileDebugJavaWithJavac UP-TO-DATE
> Task :unityLibrary:bundleLibCompileDebug UP-TO-DATE
> Task :unityLibrary:bundleLibRuntimeDebug
> Task :unityLibrary:processDebugJavaRes NO-SOURCE
> Task :unityLibrary:bundleLibResDebug UP-TO-DATE
> Task :unityLibrary:mergeDebugJniLibFolders
> Task :unityLibrary:transformNativeLibsWithMergeJniLibsForDebug
> Task :unityLibrary:transformNativeLibsWithStripDebugSymbolForDebug
> Task :launcher:mergeDebugResources
> Task :launcher:javaPreCompileDebug UP-TO-DATE
> Task :launcher:createDebugCompatibleScreenManifests UP-TO-DATE
> Task :launcher:processDebugManifest
> Task :unityLibrary:transformNativeLibsWithIntermediateJniLibsForDebug
> Task :launcher:processDebugResources FAILED

Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/5.1.1/userguide/command_line_interface.html#sec:command_line_warnings
49 actionable tasks: 31 executed, 18 up-to-date

UnityEngine.GUIUtility:ProcessEvent (int,intptr)

-----------------------------------------------------------

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':launcher:processDebugResources'.
> Android resource linking failed
  C:\_PROJECTS_\LOGIKA App\Temp\gradleOut\launcher\build\intermediates\merged_manifests\debug\AndroidManifest.xml:56: AAPT: error: unexpected element <queries> found in <manifest>.
      

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 12s
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF-8

UnityEngine.GUIUtility:ProcessEvent (int,intptr)

-----------------------------------------------------------

CommandInvokationFailure: Gradle build failed. 
C:/Program Files/Unity/Hub/Editor/2019.4.34f1/Editor/Data/PlaybackEngines/AndroidPlayer\OpenJDK\bin\java.exe -classpath "C:\Program Files\Unity\Hub\Editor\2019.4.34f1\Editor\Data\PlaybackEngines\AndroidPlayer\Tools\gradle\lib\gradle-launcher-5.1.1.jar" org.gradle.launcher.GradleMain "-Dorg.gradle.jvmargs=-Xmx4096m" "assembleDebug"

stderr[

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':launcher:processDebugResources'.
> Android resource linking failed
  C:\_PROJECTS_\LOGIKA App\Temp\gradleOut\launcher\build\intermediates\merged_manifests\debug\AndroidManifest.xml:56: AAPT: error: unexpected element <queries> found in <manifest>.
      

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 12s
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF-8
]
stdout[
> Task :launcher:preBuild UP-TO-DATE
> Task :unityLibrary:preBuild UP-TO-DATE
> Task :unityLibrary:FirebaseApp.androidlib:preBuild UP-TO-DATE
> Task :unityLibrary:FirebaseApp.androidlib:preDebugBuild UP-TO-DATE
> Task :unityLibrary:preDebugBuild UP-TO-DATE
> Task :launcher:prepareLintJar UP-TO-DATE
> Task :unityLibrary:checkDebugManifest UP-TO-DATE
> Task :unityLibrary:FirebaseApp.androidlib:checkDebugManifest
> Task :unityLibrary:processDebugManifest
> Task :unityLibrary:FirebaseApp.androidlib:processDebugManifest
> Task :unityLibrary:packageDebugRenderscript NO-SOURCE
> Task :unityLibrary:FirebaseApp.androidlib:compileDebugAidl NO-SOURCE
> Task :unityLibrary:generateDebugBuildConfig UP-TO-DATE
> Task :unityLibrary:FirebaseApp.androidlib:packageDebugRenderscript NO-SOURCE
> Task :unityLibrary:compileDebugAidl NO-SOURCE
> Task :unityLibrary:FirebaseApp.androidlib:compileDebugRenderscript NO-SOURCE
> Task :unityLibrary:compileDebugRenderscript NO-SOURCE
> Task :unityLibrary:generateDebugResValues UP-TO-DATE
> Task :unityLibrary:generateDebugResources UP-TO-DATE
> Task :unityLibrary:FirebaseApp.androidlib:generateDebugResValues
> Task :unityLibrary:FirebaseApp.androidlib:generateDebugResources
> Task :unityLibrary:packageDebugResources UP-TO-DATE
> Task :unityLibrary:prepareLintJar UP-TO-DATE
> Task :unityLibrary:generateDebugSources UP-TO-DATE
> Task :unityLibrary:mergeDebugShaders UP-TO-DATE
> Task :unityLibrary:compileDebugShaders UP-TO-DATE
> Task :unityLibrary:generateDebugAssets UP-TO-DATE
> Task :unityLibrary:FirebaseApp.androidlib:packageDebugResources
> Task :unityLibrary:FirebaseApp.androidlib:generateDebugRFile
> Task :unityLibrary:FirebaseApp.androidlib:generateDebugBuildConfig
> Task :unityLibrary:FirebaseApp.androidlib:prepareLintJar UP-TO-DATE
> Task :unityLibrary:FirebaseApp.androidlib:generateDebugSources
> Task :unityLibrary:FirebaseApp.androidlib:javaPreCompileDebug
> Task :unityLibrary:FirebaseApp.androidlib:compileDebugJavaWithJavac
> Task :unityLibrary:packageDebugAssets
> Task :unityLibrary:FirebaseApp.androidlib:bundleLibCompileDebug
> Task :unityLibrary:FirebaseApp.androidlib:mergeDebugShaders
> Task :unityLibrary:FirebaseApp.androidlib:compileDebugShaders
> Task :unityLibrary:FirebaseApp.androidlib:generateDebugAssets
> Task :unityLibrary:FirebaseApp.androidlib:packageDebugAssets
> Task :unityLibrary:FirebaseApp.androidlib:bundleLibRuntimeDebug
> Task :unityLibrary:FirebaseApp.androidlib:processDebugJavaRes NO-SOURCE
> Task :unityLibrary:FirebaseApp.androidlib:bundleLibResDebug
> Task :unityLibrary:FirebaseApp.androidlib:mergeDebugJniLibFolders
> Task :unityLibrary:FirebaseApp.androidlib:transformNativeLibsWithMergeJniLibsForDebug
> Task :unityLibrary:FirebaseApp.androidlib:transformNativeLibsWithStripDebugSymbolForDebug
> Task :unityLibrary:FirebaseApp.androidlib:transformNativeLibsWithIntermediateJniLibsForDebug
> Task :launcher:preDebugBuild
> Task :launcher:compileDebugAidl NO-SOURCE
> Task :launcher:compileDebugRenderscript NO-SOURCE
> Task :launcher:checkDebugManifest UP-TO-DATE
> Task :launcher:generateDebugBuildConfig UP-TO-DATE
> Task :launcher:generateDebugSources UP-TO-DATE
> Task :launcher:mainApkListPersistenceDebug UP-TO-DATE
> Task :launcher:generateDebugResValues UP-TO-DATE
> Task :launcher:generateDebugResources UP-TO-DATE
> Task :unityLibrary:generateDebugRFile
> Task :unityLibrary:javaPreCompileDebug
> Task :unityLibrary:compileDebugJavaWithJavac UP-TO-DATE
> Task :unityLibrary:bundleLibCompileDebug UP-TO-DATE
> Task :unityLibrary:bundleLibRuntimeDebug
> Task :unityLibrary:processDebugJavaRes NO-SOURCE
> Task :unityLibrary:bundleLibResDebug UP-TO-DATE
> Task :unityLibrary:mergeDebugJniLibFolders
> Task :unityLibrary:transformNativeLibsWithMergeJniLibsForDebug
> Task :unityLibrary:transformNativeLibsWithStripDebugSymbolForDebug
> Task :launcher:mergeDebugResources
> Task :launcher:javaPreCompileDebug UP-TO-DATE
> Task :launcher:createDebugCompatibleScreenManifests UP-TO-DATE
> Task :launcher:processDebugManifest
> Task :unityLibrary:transformNativeLibsWithIntermediateJniLibsForDebug
> Task :launcher:processDebugResources FAILED

Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/5.1.1/userguide/command_line_interface.html#sec:command_line_warnings
49 actionable tasks: 31 executed, 18 up-to-date
]
exit code: 1
UnityEditor.Android.Command.WaitForProgramToRun (UnityEditor.Utils.Program p, UnityEditor.Android.Command+WaitingForProcessToExit waitingForProcessToExit, System.String errorMsg) (at <a62274e3cf264793ab003fc3e81fb523>:0)
UnityEditor.Android.Command.Run (System.Diagnostics.ProcessStartInfo psi, UnityEditor.Android.Command+WaitingForProcessToExit waitingForProcessToExit, System.String errorMsg) (at <a62274e3cf264793ab003fc3e81fb523>:0)
UnityEditor.Android.Command.Run (System.String command, System.String args, System.String workingdir, UnityEditor.Android.Command+WaitingForProcessToExit waitingForProcessToExit, System.String errorMsg) (at <a62274e3cf264793ab003fc3e81fb523>:0)
UnityEditor.Android.AndroidJavaTools.RunJava (System.String args, System.String workingdir, System.Action`1[T] progress, System.String error) (at <a62274e3cf264793ab003fc3e81fb523>:0)
UnityEditor.Android.GradleWrapper.Run (UnityEditor.Android.AndroidJavaTools javaTools, System.String workingdir, System.String task, System.Action`1[T] progress) (at <a62274e3cf264793ab003fc3e81fb523>:0)
Rethrow as GradleInvokationException: Gradle build failed
UnityEditor.Android.GradleWrapper.Run (UnityEditor.Android.AndroidJavaTools javaTools, System.String workingdir, System.String task, System.Action`1[T] progress) (at <a62274e3cf264793ab003fc3e81fb523>:0)
UnityEditor.Android.PostProcessor.Tasks.BuildGradleProject.Execute (UnityEditor.Android.PostProcessor.PostProcessorContext context) (at <a62274e3cf264793ab003fc3e81fb523>:0)
UnityEditor.Android.PostProcessor.PostProcessRunner.RunAllTasks (UnityEditor.Android.PostProcessor.PostProcessorContext context) (at <a62274e3cf264793ab003fc3e81fb523>:0)
Rethrow as BuildFailedException: Exception of type 'UnityEditor.Build.BuildFailedException' was thrown.
UnityEditor.Android.PostProcessor.CancelPostProcess.AbortBuild (System.String title, System.String message, System.Exception ex) (at <a62274e3cf264793ab003fc3e81fb523>:0)
UnityEditor.Android.PostProcessor.PostProcessRunner.RunAllTasks (UnityEditor.Android.PostProcessor.PostProcessorContext context) (at <a62274e3cf264793ab003fc3e81fb523>:0)
UnityEditor.Android.PostProcessAndroidPlayer.PostProcess (UnityEditor.BuildTarget target, System.String stagingAreaData, System.String stagingArea, System.String playerPackage, System.String installPath, System.String companyName, System.String productName, UnityEditor.BuildOptions options, UnityEditor.RuntimeClassRegistry usedClassRegistry, UnityEditor.Build.Reporting.BuildReport report) (at <a62274e3cf264793ab003fc3e81fb523>:0)
UnityEditor.Android.AndroidBuildPostprocessor.PostProcess (UnityEditor.Modules.BuildPostProcessArgs args, UnityEditor.BuildProperties& outProperties) (at <a62274e3cf264793ab003fc3e81fb523>:0)
UnityEditor.PostprocessBuildPlayer.Postprocess (UnityEditor.BuildTargetGroup targetGroup, UnityEditor.BuildTarget target, System.String installPath, System.String companyName, System.String productName, System.Int32 width, System.Int32 height, UnityEditor.BuildOptions options, UnityEditor.RuntimeClassRegistry usedClassRegistry, UnityEditor.Build.Reporting.BuildReport report) (at <785fa035f6c842aea795e821898770c6>:0)
UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr)

-----------------------------------------------------------

Build completed with a result of 'Failed' in 212 seconds (211725 ms)
UnityEngine.GUIUtility:ProcessEvent (int,intptr)

-----------------------------------------------------------

UnityEditor.BuildPlayerWindow+BuildMethodException: 4 errors
  at UnityEditor.BuildPlayerWindow+DefaultBuildMethods.BuildPlayer (UnityEditor.BuildPlayerOptions options) [0x002bf] in <785fa035f6c842aea795e821898770c6>:0 
  at UnityEditor.BuildPlayerWindow.CallBuildMethods (System.Boolean askForBuildLocation, UnityEditor.BuildOptions defaultBuildOptions) [0x00080] in <785fa035f6c842aea795e821898770c6>:0 
UnityEngine.GUIUtility:ProcessEvent (int,intptr)

**Comment by yasirkula** [comment, marked as answer]
Hi, can you check out the relevant entry in FAQ?

**Reply by UmberFunk** [reply]
Oh, missed that... Sorry. THANKS A LOT!
